features/unit/abijson.feature

```diff
@@ -86,3 +86,30 @@ Feature: AbiJson
     Examples:
       | methodsig                | name            | description                 | network1                                     | network1-app-id | network2                                     | network2-app-id | jsonfile      | directory         |
       | add(uint32,uint32)uint32 | ExampleContract | This is an example contract | wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8= | 1234            | SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI= | 5678            | contract.json | abi_responsejsons |
+
+
+  Scenario Outline: Get a method by name from Interface
+    When I create the Method object from method signature "<methodsig>"
+    And I create another Method object from method signature "<extramethod>"
+    And I create an Interface object from the Method object
+    And I get the method from the Interface by name "<name>"
+    Then the produced method signature should equal "<methodsig>". If there is an error it begins with "<error>"
+
+    Examples:
+      | methodsig                | extramethod              | name | error           |
+      | add(uint32,uint32)uint32 |                          | add  |                 |
+      | add(uint32,uint32)uint32 |                          | sub  | found 0 methods |
+      | add(uint32,uint32)uint32 | add(uint64,uint64)uint64 | add  | found 2 methods |
+
+  Scenario Outline: Get a method by name from Contract
+    When I create the Method object from method signature "<methodsig>"
+    And I create another Method object from method signature "<extramethod>"
+    And I create a Contract object from the Method object
+    And I get the method from the Contract by name "<name>"
+    Then the produced method signature should equal "<methodsig>". If there is an error it begins with "<error>"
```

Comment:
This step suggests that our ABI Contract code has a bug in it. How can you even get to this point? Shouldn't we blow up as soon as we attempt to construct a contract with a duplicate error name or add such a duplicate?

@algochoi - any thoughts?

Reply:
dupe name is fine since it generates a diff signature (assuming args are different)

Reply:
Thinking more... overloading seems to be ok according to ARC-4. The only issue would be if the signatures have the same selector (which in practice means they have the same signature). So I guess this is ok, but then we should allow querying a method by its signature, no?

Reply:
we _could_ but I don't expect it will be common. And someone can implement their own method getter thing since the method list is public in all interface/contracts.

Reply:
> So I guess this is ok, but then we should allow querying a method by its signature, no?

I think this also makes sense. But if we don't see it being a common use case, the current approach seems fine.